tests/src/python/provider_python.py

```diff
@@ -92,7 +92,7 @@ def fetchFeature(self, f):
                 _f = self._source._features[list(self._source._features.keys())[self._index]]
                 self._index += 1
 
-                if self._feature_id_list is not None and f.id() not in self._feature_id_list:
+                if self._feature_id_list is not None and _f.id() not in self._feature_id_list:
```

Comment:
Iterating over the wrong feature, causing `f.id()` to always return 0, effectively causing this test to always return True.